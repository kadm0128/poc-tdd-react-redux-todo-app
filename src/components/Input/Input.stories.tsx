import type { Meta } from '@storybook/react';
import InputComponent from './Input';

type Props = {
  color?: 'primary' | 'secondary' | 'success' | 'info' | 'warning' | 'error';
  variant?: 'filled' | 'outlined'| 'standard';
  size?: 'small' | 'medium' | 'large';
  className?: string;
  label?: string;
};

const meta = {
  title: 'Example/Input',
  component: InputComponent,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    color: {
      control: 'select',
      options: ['primary', 'secondary', 'success', 'info', 'warning', 'error'],
    },
  },
} satisfies Meta<typeof InputComponent>;

export default meta;

export const Input = (props: Props) => <InputComponent {...props} />
