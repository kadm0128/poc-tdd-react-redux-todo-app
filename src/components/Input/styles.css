.dc-wrapper {
   display: flex;
   flex-direction: column;
}

.dc-label-wrapper {
   display: flex;
   align-items: center;
}

.dc-input-wrapper {
   display: flex;
   margin-top: 0.75rem;
}

.dc-form .dc-input-wrapper {
   margin-top: 0;
}

.dc-form-vertical > * + * {
   margin-top: 0.75rem;
}

.dc-form-horizontal > * + * {
   margin-top: 0.75rem;
}

.dc-form-horizontal .dc-wrapper {
   flex-direction: row;
}

.dc-form-vertical .dc-wrapper {
   flex-direction: column;
}

.dc-form-inline .dc-wrapper {
   flex-direction: row;
}

.dc-form-horizontal .dc-label-wrapper {
   margin-right: 1rem;
}

.dc-form-inline .dc-label-wrapper {
   margin-right: 1rem;
}

.dc-form-inline .dc-input-wrapper {
   margin-right: 1rem;
}

.dc-label {
   display: inline-block;
   font-family: "Roboto", "Helvetica", "Arial", sans-serif;
   font-weight: 400;
   line-height: 1.5;
   text-decoration: none;
}

.dc-input {
   display: inline-block;
   font-family: "Roboto", "Helvetica", "Arial", sans-serif;
   font-weight: 400;
   line-height: 1.5;
   text-decoration: none;
   vertical-align: middle;
}

.dc-form-horizontal .dc-label,
.dc-form-horizontal .dc-input,
.dc-form-vertical .dc-label,
.dc-form-vertical .dc-input {
   display: flex;
   flex-grow: 1;
}

.dc-input-filled-primary {
   color: white;
   background-color: #1976d2;
   border: 1px solid #1565c0;
}

.dc-input-filled-primary:hover {
   color: white;
   background-color: rgb(21, 101, 192);
   box-shadow: 0 0 0 1px rgba(21, 101, 192, 0.102);
}

.dc-input-filled-primary:focus {
   color: white;
   background-color: #1565c0;
   box-shadow: 0 0 0 2px rgba(21, 101, 192, 0.2);
   outline: none;
}

.dc-input-outlined-primary {
   background-color: transparent;
   border: 1px solid #1976d2;
}

.dc-input-outlined-primary:hover {
   border: 1px solid #1565c0;
   box-shadow: 0 0 0 1px rgba(21, 101, 192, 0.102);
}

.dc-input-outlined-primary:focus {
   border: 1px solid #1565c0;
   box-shadow: 0 0 0 2px rgba(21, 101, 192, 0.2);
   outline: none;
}

.dc-input-standard-primary {
   color: #1976d2;
   background-color: transparent;
   border: none;
}

.dc-input-standard-primary:hover {
   color: #1565c0;
   background-color: #f0f7fd;
   border: none;
}

.dc-input-standard-primary:focus {
   color: #1565c0;
   background-color: #f0f7fd;
   border: none;
   outline: none;
}

.dc-input-filled-secondary {
   color: white;
   background-color: #9c27b0;
   border: 1px solid #7b1fa2;
}

.dc-input-filled-secondary:hover {
   color: white;
   background-color: rgb(123, 31, 162);
   box-shadow: 0 0 0 1px rgba(123, 31, 162, 0.102);
}

.dc-input-filled-secondary:focus {
   color: white;
   background-color: rgb(123, 31, 162);
   box-shadow: 0 0 0 1px rgba(123, 31, 162, 0.102);
   outline: none;
}

.dc-input-outlined-secondary {
   background-color: transparent;
   border: 1px solid #9c27b0;
}

.dc-input-outlined-secondary:hover {
   border: 1px solid rgb(123, 31, 162);
   box-shadow: 0 0 0 1px rgba(123, 31, 162, 0.102);
}

.dc-input-outlined-secondary:focus {
   border: 1px solid rgb(123, 31, 162);
   box-shadow: 0 0 0 1px rgba(123, 31, 162, 0.102);
   outline: none;
}

.dc-input-standard-secondary {
   color: #9c27b0;
   background-color: transparent;
   border: none;
}

.dc-input-standard-secondary:hover {
   color: #7b1fa2;
   background-color: #faf0fc;
   border: none;
}

.dc-input-standard-secondary:focus {
   color: #7b1fa2;
   background-color: #faf0fc;
   border: none;
   outline: none;
}

.dc-input-filled-success {
   color: white;
   background-color: #2e7d32;
   border: 1px solid #1b5e20;
}

.dc-input-filled-success:hover {
   color: white;
   background-color: rgb(27, 94, 32);
   box-shadow: 0 0 0 1px rgba(27, 94, 32, 0.102);
}

.dc-input-filled-success:focus {
   color: white;
   background-color: rgb(27, 94, 32);
   box-shadow: 0 0 0 1px rgba(27, 94, 32, 0.102);
   outline: none;
}

.dc-input-outlined-success {
   background-color: transparent;
   border: 1px solid #2e7d32;
}

.dc-input-outlined-success:hover {
   border: 1px solid rgb(27, 94, 32);
   box-shadow: 0 0 0 1px rgba(27, 94, 32, 0.102);
}

.dc-input-outlined-success:focus {
   border: 1px solid rgb(27, 94, 32);
   box-shadow: 0 0 0 1px rgba(27, 94, 32, 0.102);
   outline: none;
}

.dc-input-standard-success {
   color: #2e7d32;
   background-color: transparent;
   border: none;
}

.dc-input-standard-success:hover {
   color: #1b5e20;
   background-color: #f0faf1;
   border: none;
}

.dc-input-standard-success:focus {
   color: #1b5e20;
   background-color: #f0faf1;
   border: none;
   outline: none;
}

.dc-input-filled-info {
   color: white;
   background-color: #0288d1;
   border: 1px solid #01579b;
}

.dc-input-filled-info:hover {
   color: white;
   background-color: rgb(1, 87, 155);
   box-shadow: 0 0 0 1px rgba(1, 87, 155, 0.102);
}

.dc-input-filled-info:focus {
   color: white;
   background-color: rgb(1, 87, 155);
   box-shadow: 0 0 0 1px rgba(1, 87, 155, 0.102);
   outline: none;
}

.dc-input-outlined-info {
   background-color: transparent;
   border: 1px solid #0288d1;
}

.dc-input-outlined-info:hover {
   border: 1px solid #01579b;
   box-shadow: 0 0 0 1px rgba(1, 87, 155, 0.102);
}

.dc-input-outlined-info:focus {
   border: 1px solid #01579b;
   box-shadow: 0 0 0 1px rgba(1, 87, 155, 0.102);
   outline: none;
}

.dc-input-standard-info {
   color: #0288d1;
   background-color: transparent;
   border: none;
}

.dc-input-standard-info:hover {
   color: #01579b;
   background-color: #edf9ff;
   border: none;
}

.dc-input-standard-info:focus {
   color: #01579b;
   background-color: #edf9ff;
   border: none;
   outline: none;
}

.dc-input-filled-warning {
   color: white;
   background-color: #ed6c02;
   border: 1px solid #e65100;
}

.dc-input-filled-warning:hover {
   color: white;
   background-color: rgb(230, 81, 0);
   box-shadow: 0 0 0 1px rgba(230, 81, 0, 0.102);
}

.dc-input-filled-warning:focus {
   color: white;
   background-color: rgb(230, 81, 0);
   box-shadow: 0 0 0 1px rgba(230, 81, 0, 0.102);
   outline: none;
}

.dc-input-outlined-warning {
   background-color: transparent;
   border: 1px solid #ed6c02;
}

.dc-input-outlined-warning:hover {
   border: 1px solid #e65100;
   box-shadow: 0 0 0 1px rgba(230, 81, 0, 0.102);
}

.dc-input-outlined-warning:focus {
   border: 1px solid #e65100;
   box-shadow: 0 0 0 1px rgba(230, 81, 0, 0.102);
   outline: none;
}

.dc-input-standard-warning {
   color: #ed6c02;
   background-color: transparent;
   border: none;
}

.dc-input-standard-warning:hover {
   color: #e65100;
   background-color: #fff6ef;
   border: none;
}

.dc-input-standard-warning:focus {
   color: #e65100;
   background-color: #fff6ef;
   border: none;
   outline: none;
}

.dc-input-filled-error {
   color: white;
   background-color: #d32f2f;
   border: 1px solid rgb(198, 40, 40);
}

.dc-input-filled-error:hover {
   color: white;
   background-color: #c62828;
   box-shadow: 0 0 0 1px rgba(198, 40, 40, 0.102);
}

.dc-input-filled-error:focus {
   color: white;
   background-color: #c62828;
   box-shadow: 0 0 0 1px rgba(198, 40, 40, 0.102);
   outline: none;
}

.dc-input-outlined-error {
   background-color: transparent;
   border: 1px solid #d32f2f;
}

.dc-input-outlined-error:hover {
   border: 1px solid #c62828;
   box-shadow: 0 0 0 1px rgba(198, 40, 40, 0.102);
}

.dc-input-outlined-error:focus {
   border: 1px solid #c62828;
   box-shadow: 0 0 0 1px rgba(198, 40, 40, 0.102);
   outline: none;
}

.dc-input-standard-error {
   color: #d32f2f;
   background-color: transparent;
   border: none;
}

.dc-input-standard-error:hover {
   color: #c62828;
   background-color: #fcf3f3;
   border: none;
}

.dc-input-standard-error:focus {
   color: #c62828;
   background-color: #fcf3f3;
   border: none;
   outline: none;
}

.dc-input-small {
   padding: 0.25rem 0.625rem;
   font-size: 0.813rem;
}

.dc-input-medium {
   padding: 0.375rem 1rem;
   font-size: 0.875rem;
}

.dc-input-large {
   padding: 0.5rem 1.375rem;
   font-size: 0.938rem;
}
