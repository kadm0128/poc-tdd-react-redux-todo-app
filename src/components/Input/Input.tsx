import { useId } from "react";
import { getClassNames } from "../../utils";
import { Color, Size, Variant } from "./Input.types";
import "./styles.css";

type Props = {
   color?: Color;
   variant?: Variant;
   size?: Size;
   inputClassName?: string;
   labelClassName?: string;
   wrapperClassName?: string;
   labelWrapperClassName?: string;
   inputWrapperClassName?: string;
   label?: string;
}

const Input = (props: Props) => {
   const { color = 'primary', variant = 'outlined', size = 'medium', inputClassName = '', labelClassName = '',
      wrapperClassName = '', labelWrapperClassName = '', inputWrapperClassName = '', ...rest } = props;
   const { label } = rest;

   const inputClassNames = getClassNames('input', [`${variant}-${color}`, size], inputClassName);
   const labelClassNames = getClassNames('label', [], labelClassName);
   const wrapperClassNames = getClassNames('wrapper', [], wrapperClassName);
   const labelWrapperClassNames = getClassNames('label-wrapper', [], labelWrapperClassName);
   const inputWrapperClassNames = getClassNames('input-wrapper', [], inputWrapperClassName);
   const id = useId();

   return (
      <div className={wrapperClassNames}>
         { label && (
            <div className={labelWrapperClassNames}>
               <label className={labelClassNames} htmlFor={id}>{label}</label>
            </div>
         )}
         <div className={inputWrapperClassNames}>
            <input className={inputClassNames} id={id} />
         </div>
      </div>
   );
}

export default Input;