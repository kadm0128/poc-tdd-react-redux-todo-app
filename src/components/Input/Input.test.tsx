import { render, screen, renderer } from "../../utils//test-utils";
import Input from "./Input";
import { Color, Size, Variant } from "./Input.types";


describe('INPUT', () => {
   describe('Component Rendering and Props Handling', () => {
      test('applies default styles and classes correctly', () => {
         const { container } = render(<Input />);
         const input = container.querySelector('input');
         expect(input).toHaveClass('dc-input dc-input-outlined-primary dc-input-medium');
      });

      test('renders the specified label correctly', () => {
         render(<Input label="First Name" />);
         const label = screen.getByText('First Name');
         expect(label).toBeInTheDocument();
      });

      describe('renders the specified color correctly', () => {
         const supportedColors: Color[] = ['primary', 'secondary', 'success', 'info', 'warning', 'error'];

         test.each(supportedColors)('renders with specified color - %s', (color: Color) => {
            const { container } = render(<Input color={color} />);
            const input = container.querySelector('input');
            expect(input).toHaveClass(`dc-input dc-input-outlined-${color} dc-input-medium`);
         });
      });

      describe('renders the specified variant correctly', () => {
         const supportedVariants: Variant[] = ['filled', 'outlined', 'standard'];

         test.each(supportedVariants)('renders with specified variant - %s', (variant: Variant) => {
            const { container } = render(<Input variant={variant} />);
            const input = container.querySelector('input');
            expect(input).toHaveClass(`dc-input dc-input-${variant}-primary dc-input-medium`);
         });
      });

      describe('renders the specified size correctly', () => {
         const supportedSizes: Size[] = ['small', 'medium', 'large'];

         test.each(supportedSizes)('renders with specified size - %s', (size: Size) => {
            const { container } = render(<Input size={size} />);
            const input = container.querySelector('input');
            expect(input).toHaveClass(`dc-input dc-input-outlined-primary dc-input-${size}`);
         });
      });

      test('renders the given set of props correctly', () => {
         const { container } = render(<Input size="small" color="success" variant="filled" />);
         const input = container.querySelector('input');
         expect(input).toHaveClass('dc-input dc-input-filled-success dc-input-small');
      });
   });

   // Test interactions within component (testing event handlers, state changes, and component updates triggered by user interactions)
   describe('Component Interactions and State Management', () => {
      test('test', () => {

      });

      test('test', () => {

      })
   });

   describe('Component Error Handling', () => {
      test('test', () => {

      })
   });

   describe('Snapshot Testing', () => {
      const supportedSizes: Size[] = ['small', 'medium', 'large'];
      const supportedVariants: Variant[] = ['filled', 'outlined', 'standard'];
      const supportedColors: Color[] = ['primary', 'secondary', 'success', 'info', 'warning', 'error'];

      for (const color of supportedColors) {
         for (const variant of supportedVariants) {
            for (const size of supportedSizes) {
               test(`Snapshot Testing - ${color}, ${variant}, ${size}`, () => {
                  const tree = renderer.create(<Input size={size} color={color} variant={variant} />).toJSON();
                  expect(tree).toMatchSnapshot();
               });
            }
         }
      }
   });

   describe('Integration Testing', () => {
      test('test', () => {

      })
   });

   describe('Accessibility', () => {
      test('test', () => {

      })
   });

   describe('Edge Cases', () => {
      test('test', () => {

      })
   });
})