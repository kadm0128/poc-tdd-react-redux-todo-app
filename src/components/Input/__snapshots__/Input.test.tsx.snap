// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`INPUT > Snapshot Testing > Snapshot Testing - error, filled, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-error dc-input-large"
      id=":r1f:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - error, filled, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-error dc-input-medium"
      id=":r1e:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - error, filled, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-error dc-input-small"
      id=":r1d:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - error, outlined, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-error dc-input-large"
      id=":r1i:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - error, outlined, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-error dc-input-medium"
      id=":r1h:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - error, outlined, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-error dc-input-small"
      id=":r1g:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - error, standard, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-error dc-input-large"
      id=":r1l:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - error, standard, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-error dc-input-medium"
      id=":r1k:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - error, standard, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-error dc-input-small"
      id=":r1j:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - info, filled, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-info dc-input-large"
      id=":rt:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - info, filled, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-info dc-input-medium"
      id=":rs:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - info, filled, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-info dc-input-small"
      id=":rr:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - info, outlined, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-info dc-input-large"
      id=":r10:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - info, outlined, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-info dc-input-medium"
      id=":rv:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - info, outlined, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-info dc-input-small"
      id=":ru:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - info, standard, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-info dc-input-large"
      id=":r13:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - info, standard, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-info dc-input-medium"
      id=":r12:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - info, standard, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-info dc-input-small"
      id=":r11:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - primary, filled, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-primary dc-input-large"
      id=":r2:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - primary, filled, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-primary dc-input-medium"
      id=":r1:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - primary, filled, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-primary dc-input-small"
      id=":r0:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - primary, outlined, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-primary dc-input-large"
      id=":r5:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - primary, outlined, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-primary dc-input-medium"
      id=":r4:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - primary, outlined, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-primary dc-input-small"
      id=":r3:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - primary, standard, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-primary dc-input-large"
      id=":r8:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - primary, standard, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-primary dc-input-medium"
      id=":r7:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - primary, standard, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-primary dc-input-small"
      id=":r6:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - secondary, filled, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-secondary dc-input-large"
      id=":rb:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - secondary, filled, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-secondary dc-input-medium"
      id=":ra:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - secondary, filled, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-secondary dc-input-small"
      id=":r9:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - secondary, outlined, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-secondary dc-input-large"
      id=":re:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - secondary, outlined, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-secondary dc-input-medium"
      id=":rd:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - secondary, outlined, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-secondary dc-input-small"
      id=":rc:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - secondary, standard, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-secondary dc-input-large"
      id=":rh:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - secondary, standard, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-secondary dc-input-medium"
      id=":rg:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - secondary, standard, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-secondary dc-input-small"
      id=":rf:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - success, filled, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-success dc-input-large"
      id=":rk:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - success, filled, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-success dc-input-medium"
      id=":rj:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - success, filled, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-success dc-input-small"
      id=":ri:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - success, outlined, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-success dc-input-large"
      id=":rn:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - success, outlined, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-success dc-input-medium"
      id=":rm:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - success, outlined, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-success dc-input-small"
      id=":rl:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - success, standard, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-success dc-input-large"
      id=":rq:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - success, standard, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-success dc-input-medium"
      id=":rp:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - success, standard, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-success dc-input-small"
      id=":ro:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - warning, filled, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-warning dc-input-large"
      id=":r16:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - warning, filled, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-warning dc-input-medium"
      id=":r15:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - warning, filled, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-filled-warning dc-input-small"
      id=":r14:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - warning, outlined, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-warning dc-input-large"
      id=":r19:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - warning, outlined, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-warning dc-input-medium"
      id=":r18:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - warning, outlined, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-outlined-warning dc-input-small"
      id=":r17:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - warning, standard, large 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-warning dc-input-large"
      id=":r1c:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - warning, standard, medium 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-warning dc-input-medium"
      id=":r1b:"
    />
  </div>
</div>
`;

exports[`INPUT > Snapshot Testing > Snapshot Testing - warning, standard, small 1`] = `
<div
  className="dc-wrapper"
>
  <div
    className="dc-input-wrapper"
  >
    <input
      className="dc-input dc-input-standard-warning dc-input-small"
      id=":r1a:"
    />
  </div>
</div>
`;
